simulator.py

```diff
@@ -2,33 +2,17 @@
 Simulator that simulates trading process and estimates profits and losses.
 """
 
-import copy
 import uuid
 from thousandaire.data_classes import Data
 from thousandaire.constants import TRADING_INSTRUMENTS
 
-def decode_data(data):
-    """
-    Decode data shared by the simulation main process.
-
-    In multi-processing, the simulator will get data by shared memory, which
-    are encoded by pickle to make them sharable.
-    To avoid polluting the shared memory, we need to create a local copy.
-    Since 'price' may also be in 'others', we should deepcopy the data
-    separately instead of using copy.deepcopy(data).
-    """
-    return {
-        'workdays' : copy.deepcopy(data['workdays']),
-        'price' : copy.deepcopy(data['price']),
-        'others' : copy.deepcopy(data['others'])}
-

```

Comment:
All the data will come from data server in the future, so I move this into the data part in simulation.py and combine it in extract_data.